modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -365,7 +371,10 @@ protected HttpMessage createMessage(String[] initialLine) throws Exception {
                     );
             }
             // combines the HTTP message pieces into a single full HTTP request (with headers and body)
-            final HttpObjectAggregator aggregator = new Netty4HttpAggregator(handlingSettings.maxContentLength());
+            final HttpObjectAggregator aggregator = new Netty4HttpAggregator(
+                handlingSettings.maxContentLength(),
+                httpPreRequest -> enabled.incrementalBulkEnabled() && httpPreRequest.uri().contains("_bulk") == false
```

Comment:
I still need to fix this on my side. For my integ tests I need to inject decider that takes testing uri. With this change my integ test will not work.

Reply:
For now, to unblock current change would you mind to add this condition?
`req.uri().startsWith("/_test/request-stream") == false || (enabled.incrementalBulkEnabled() && httpPreRequest.uri().contains("_bulk") == false)`

Reply:
I have few drafts how to do this. But put it aside in favour of removing http-object-aggregator. It seems need to continue with injection, removal is a big change.

Reply:
Can I just do this:

```java
    @Override
    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
        assert msg instanceof HttpObject;
        if (msg instanceof HttpRequest request) {
            var preReq = HttpHeadersAuthenticatorUtils.asHttpPreRequest(request);
            aggregating = decider.test(preReq) && IGNORE_TEST.test(preReq);
        }
        if (aggregating || msg instanceof FullHttpRequest) {
            super.channelRead(ctx, msg);
        } else {
            handle(ctx, (HttpObject) msg);
        }
    }
```

Reply:
nice!